crates/engine/primitives/src/config.rs

```diff
@@ -79,6 +79,8 @@ pub struct TreeConfig {
     precompile_cache_enabled: bool,
     /// Whether to use state root fallback for testing
     state_root_fallback: bool,
+    /// Whether to force returning the payload on forkchoice updated, even if head is canonical.
+    force_payload_return_on_fcu: bool,
```

Comment:
I'd like to rename this to something like

```
process_payload_attributes_on_canonical_head
```

even if more verbose this is makes a bit clearer what this setting is for.
we can maybe shorten this for the cli arg.

I'd also like to reference the engine api rule and why this setting exists.

Reply:
I've updated the code with the new name `process_payload_attributes_on_canonical_head` and added comments throughout. Could you clarify which additional documentation you'd like me to include?